lib/jvm.js

```diff
@@ -395,6 +407,43 @@ class JvmMethodMangler {
   }
 }
 
+function doManglers (vm) {
+  vm.perform(() => {
+    const api = getApi();
+    const env = vm.getEnv();
+    const thread = api['JavaThread::thread_from_jni_environment'](env.handle);
+    let force = false;
+    withJvmThread(() => {
+      replaceManglers.forEach(mangler => {
+        const { method, originalMethod, impl, methodId, newMethod } = mangler;
+        if (originalMethod === null) {
+          mangler.originalMethod = fetchJvmMethod(method);
+          mangler.newMethod = nativeJvmMethod(method, impl, thread);
+          installJvmMethod(mangler.newMethod, methodId, thread);
+        } else {
+          api['Method::set_native_function'](newMethod.method, impl, 0);
+        }
+      });
+      revertManglers.forEach(mangler => {
+        const { originalMethod, methodId, newMethod } = mangler;
+        if (originalMethod !== null) {
+          revertJvmMethod(originalMethod);
+          const revert = originalMethod.oldMethod;
+          revert.oldMethod = newMethod;
+          installJvmMethod(revert, methodId, thread);
+          force = true;
+        }
+      });
+    });
+    if (force) {
+      forceSweep(env.handle);
+    }
+    replaceManglers.clear();
+    revertManglers.clear();
+    manglersScheduled = false;
```

Comment:
I think we need to do these operations first thing, because any `NativeFunction`-call with `scheduling: 'cooperative'`, which is the default, will yield the JS lock, so another thread may in theory run JS, and start manipulating these data structures.

Reply:
I see what you're saying and I adjusted the call order in next commit